code/clue.js

```diff
@@ -1,59 +1,350 @@
-  // ...  and the rest
-]
+  professorPlum,
+  missScarlet,
+  mrsPeackok,
+  colonelMustard,
+  mrsWhite,
+];
+
+const weapons = [
+  rope,
+  knife,
+  candlestick,
+  dumbbell,
+  poison,
+  axe,
+  bat,
+  trophy,
+  pistol,
+];
+
+const rooms = [
+  "Dining Room",
+  "Conservatory",
+  "Kitchen",
+  "Study",
+  "Library",
+  "Billiard Room",
+  "Lounge",
+  "Ballroom",
+  "Hall",
+  "Spa",
+  "Living Room",
+  "Observatory",
+  "Theater",
+  "Guest House",
+  "Patio",
+];
 
-const weapons = []
+console.log(suspects);
+console.log(suspects[3].occupation);
+console.log(weapons);
+console.log(rooms.indexOf("Spa"));
 
-const rooms = []
+const weaponRooms = {
+  rope: ["Conservatory", "Hall", "Theater", "Patio", "Observatory"],
+  knife: ["Kitchen"],
+  candlestick: [
+    "Dining Room",
+    "Conservatory",
+    "Study",
+    "Library",
+    "Lounge",
+    "Ballroom",
+    "Hall",
+    "Spa",
+    "Living Room",
+    "Guest House",
+    "Patio",
+    "Theater",
+  ],
+  dumbbell: ["Guest House"],
+  poison: [
+    "Dining Room",
+    "Conservatory",
+    "Kitchen",
+    "Living Room",
+    "Guest House",
+    "Patio",
+  ],
+  axe: ["Conservatory", "Billiard Room", "Hall", "Patio"],
+  bat: ["Billiard Room", "Guest House"],
+  trophy: ["Living Room", "Guest House"],
+  pistol: ["Living Room", "Gest House"],
+};
 
 // THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT YOU PASS IN TO THE FUNCTION.
 // YOU DON'T NEED TO CHANGE THIS, JUST TRY TO UNDERSTAND IT. AND HOW TO USE IT.
-const randomSelector = array => {
-  return array[Math.floor(Math.random() * array.length)]
-}
+
+const randomSelector = (array) => {
+  return array[Math.floor(Math.random() * array.length)];
+};
 
 // CREATE AN OBJECT THAT KEEPS THE MYSTERY.
 // With a killer, a weapon and a room.
 // The values will be set later.
+const mystery = {
+  killer: undefined,
+  weapon: undefined,
+  room: undefined,
+  favWeapon: undefined,
+};
+
+// shuffleFavouriteWeapon function
+const duplicateItems = (arr) => {
+  let newArr = [];
+  for (let i = 0; i < 10; i++) {
+    newArr = newArr.concat(arr);
+  }
+  return newArr;
+};
+
+const favWeapons = [rope, knife, candlestick, dumbbell, poison, axe];
+const shuffleFavouriteWeapon = () => {
+  mystery.favWeapon = shuffle(favWeapons, weapons);
+  document.getElementById(
+    "mysteryName"
+  ).innerHTML = `Favorite wepon is ${mystery.favWeapon.id}`;
+};
+
+const shuffle = (favArr, arr) => {
+  const favDuplicated = duplicateItems(favArr);
+  const concatArr = [...arr, ...favDuplicated];
+  return randomSelector(concatArr);
+};
+// ----------------WEAPON AND ROOM
+
+const shuffleFavRoom = () => {
+  if (mystery.weapon !== undefined) {
+    const roomList = weaponRooms[mystery.weapon.id];
+    mystery.room = shuffle(roomList, rooms);
+    document.getElementById("roomName").innerHTML = mystery.room;
+  } else {
+    mystery.room = randomSelector(rooms);
+    document.getElementById("roomName").innerHTML = mystery.room;
+  }
+};
```

Comment:
This seems to be a really smooth way of achieving the goal, having a certain weapon more likely to a appear in certain rooms. Great job! 💯